packages/cozy-client/src/CozyClient.js

```diff
@@ -834,6 +834,14 @@ client.query(Q('io.cozy.bills'))`)
         return
       }
     }
+    // If we already have the same query in loading state, no
+    // need to refetch it. We return the promise we already
+    // have in the promiseCache
+    if (existingQuery && Object.keys(existingQuery).length > 0) {
```

Comment:
is `Object.keys(existingQuery).length > 0` necessary as we check if `existingQuery.fetchStatus` has a value on the next line?

Reply:
It's a good question. I think it can't be happen in real live to have `existingQuery` something else than `null` or `{}`. 

Replaced by  
```
if (existingQuery && existingQuery.fetchStatus === 'loading') { 
```